2024Robot_Team3484/src/main/include/subsystems/IntakeSubsystem.h

```diff
@@ -0,0 +1,42 @@
+#ifndef INTAKE_H
+#define INTAKE_H
+
+
+#include <frc2/command/SubsystemBase.h>
+#include <rev/CANSparkMax.h>
+//#include <ctre/Phoenix.h>
+#include <frc/DigitalInput.h>
+#include <frc/trajectory/TrapezoidProfile.h>
+
+#include <units/angle.h>
+
+#include <Constants.h>
+
+class IntakeSubsystem : public frc2::SubsystemBase {
+    public:
+        IntakeSubsystem();
+        void Periodic() override;
+        void SetIntakeAngle(units::degree_t angle);
+        void SetRollerPower(double power);
+        bool HasPiece();
+        bool ArmExtended();
+        units::turn_t GetIntakePosition();
+
+    private:
+        bool _arm_sensor_hit = false;
+        
+        rev::CANSparkMax _pivot_motor{IntakeConstants::PIVOT_MOTOR_PORT, rev::CANSparkMax::MotorType::kBrushless};
+        rev::CANSparkMax _drive_motor{IntakeConstants::DRIVE_MOTOR_PORT, rev::CANSparkMax::MotorType::kBrushed};
+
+        frc::DigitalInput _piece_sensor{IntakeConstants::PIECE_SENSOR_PORT};
+        frc::DigitalInput _arm_sensor{IntakeConstants::ARM_SENSOR_PORT};
```

Comment:
Similar comment to the CAN ID's. Pass the DI channel numbers through the constructor, don't reference them directly in the header file.